### PR TITLE
Add the minimum required to the error message, when trying to apply a cart rule with a minimum

### PR DESCRIPTION
**Pull request by Prestaplugins**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add the minimum required to the error message, when trying to apply a cart rule with a minimum
| Type?             | improvement
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     |
| How to test?      | 1. Create a voucher with a minimum amount.<br/>2. Add some product to the cart but with a cart total lower than the minimum of the voucher, so the voucher is not eligible.<br/>3. Try to apply the voucher to the cart.<br/>4. This PR adds an error message that tell the customer the minimum amount requested.
| Possible impacts? |
| Sponsor company   | Prestaplugins

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27397)

### CODE REVIEW

File: classes/CartRule.php [hunk 1]
```diff
@@ -830,7 +830,7 @@ public function checkValidity(Context $context, $alreadyInCart = false, $display
             }
 
             if ($cartTotal < $minimum_amount) {
-                return (!$display_error) ? false : $this->trans('You have not reached the minimum amount required to use this voucher', [], 'Shop.Notifications.Error');
+                return (!$display_error) ? false : $this->trans('You have not reached the %s minimum amount required to use this voucher', [$context->getCurrentLocale()->formatPrice($minimum_amount, $context->currency->iso_code)], 'Shop.Notifications.Error');
```
Comment (Julievrz) [inline classes/CartRule.php]:
```suggestion
                return (!$display_error) ? false : $this->trans('The minimum amount to benefit from this promo code is %s.', [$context->getCurrentLocale()->formatPrice($minimum_amount, $context->currency->iso_code)], 'Shop.Notifications.Error');
```

This is a great idea to add the minimum amount. The error message is more useful to the user that way. I suggest using "promo code" to stay consistent with the wording used in the front office (see screenshot). I also tried to adopt a more positive tone. :-) See the Content style guide: https://zeroheight.com/80a6c5a61/v/latest/p/510e8a-components/b/2943bd 

image

Comment (matks) [inline classes/CartRule.php]: The tests reported a usecase where this code fails

```
001 Scenario: cart rule with minimum amount doesn't apply if cart total is lower # Features/Scenario/Cart/Adding/CartRule/minimum_amount_cartrule.feature:12
      When I delete product "product1" from cart "dummy_custom_cart"             # Features/Scenario/Cart/Adding/CartRule/minimum_amount_cartrule.feature:42
        Fatal error: Call to a member function formatPrice() on null (Behat\Testwork\Call\Exception\FatalThrowableError)
```

`$context->getCurrentLocale()` can be null in some circumstances

**Review by matks — CHANGES_REQUESTED** [review]

File: classes/CartRule.php (hunk 1 above)
Comment (kpodemski) [inline classes/CartRule.php]:
```suggestion
                return (!$display_error) ? false : $this->trans('You have not reached the %s minimum amount required to use this voucher', [Tools::getContextLocale($context)->formatPrice($minimum_amount, $context->currency->iso_code)], 'Shop.Notifications.Error');
```

**Comment by kpodemski** [comment]
Hello @Prestaplugins 

Any chance for you to apply our feedback to your PR? It looks like a good improvement. It's worth having it in the core. Thank you.

**Comment by kpodemski** [comment]
Hi @Prestaplugins 

You forgot to apply the wordings recommendation.
https://capture.dropbox.com/YpOXAE0IDXBaT6Qe

**Review by Progi1984 — APPROVED** [review]

**Review by Julievrz — APPROVED** [review]
Wording ✔️

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
I improved a little the "How to test?"

**Review by kpodemski — APPROVED** [review]

**Review by AureRita — APPROVED** [review]
Hi @Prestaplugins 

Thank you for your PR, I tested it and it works, as you can see here :

https://user-images.githubusercontent.com/100699445/165546425-69bf6e47-77a2-482e-b46d-dc13c99880e5.mp4

**Comment by Progi1984** [comment]
Thanks @Prestaplugins & @AureRita

**Pull request merged**